code/modules/economy/economy_machinery/eftpos.dm

```diff
@@ -172,6 +172,9 @@
 		return
 
 	var/datum/money_account/D = GLOB.station_money_database.find_user_account(C.associated_account_number, include_departments = FALSE)
+	if(!D)
+		to_chat(user, "<span class='warning'>Your currently in use card is not connected to a money account.</span>")
+		return
```

Comment:
could you remove the null check on `L185` since you're putting this one here

Reply:
done